src/pages/issues.tsx

```diff
@@ -9,36 +8,88 @@ import {
     NO_ISSUES_FOUND_MESSAGE,
 } from '@/components/constants/messages';
 import { ISSUES_URL } from '@/components/constants/url';
+import axios from 'axios';
+
+type SearchFieldProps = {
+    searchText: string;
+    onSearchTextChanged: (event: any) => void;
+    onSearchTextSubmitted: () => void;
+    loading: boolean;
+};
+
+const SearchField = ({
+    searchText,
+    onSearchTextChanged,
+    onSearchTextSubmitted,
+    loading,
+}: SearchFieldProps) => {
+    return (
+        <div className={classNames.searchFieldContainer}>
+            <input
+                placeholder="Enter query string to search issues"
+                value={searchText}
+                onChange={onSearchTextChanged}
+                className={classNames.issueSearchInput}
+            />
+            <button
+                onClick={onSearchTextSubmitted}
+                className={classNames.issuesSearchSubmitButton}
+                disabled={loading || !searchText.trim()}
+            >
+                Submit
+            </button>
+        </div>
+    );
+};
 
 const Issues: FC = () => {
     const [issueList, setIssueList] = useState<[]>([]);
+    const [searchText, setSearchText] = useState('');
+    const [isLoading, setIsLoading] = useState(true);
+    const [error, setError] = useState<null | any>(null);
 
-    const { response, error, isLoading } = useFetch(ISSUES_URL, {
-        data: {
-            userData: {
-                roles: {
-                    super_user: true,
-                },
-            },
-        },
-    });
+    const fetchIssues = async () => {
+        try {
+            setIsLoading(true);
+            const res = await fetch(`${ISSUES_URL}?q=${searchText}`);
+            const data = await res.json();
+            if ('issues' in data) {
+                const issuesAndPullRequests: any = data.issues;
+                const onlyIssues = issuesAndPullRequests.filter(
+                    (issue: { hasOwnProperty: (arg0: string) => any }) =>
+                        !Object.prototype.hasOwnProperty.call(
+                            issue,
+                            'pull_request'
+                        )
+                );
+                setIssueList(onlyIssues);
+            }
+            setIsLoading(false);
+            setError(null);
+        } catch (error) {
```

Comment:
also add a `console.error` here

Reply:
Done. 
But can you please specify why it's needed? Because I've been always told to avoid adding any console messages and should not be pushed to prod.

Reply:
so when you set up error logging tools like sentry, crashlytics and etc if you miss adding `console.error` it doesnt properly catch your stack trace sometimes also it becomes easier for these tools as they capture all sorts of logs which have errors. Hence, you must throw &|| use console.error where ever it is needed

Reply:
Also, the statement which is told to you is partially true here is one example of what I am talking 
you must not add `console.logs`  as you might render out some crucial data in console,
 but when you have errors as I mentioned in the above comment why everyone must do it right. Below is an SS of the current pull request page dev console.
![image](https://user-images.githubusercontent.com/64726586/236108726-ae3cc7d4-ad06-495d-9cd4-5d50464e9cef.png)

Reply:
Got it thank you!